petab/problem.py

```diff
@@ -347,66 +347,111 @@ def get_noise_distributions(self):
-        return list(self.parameter_df[NOMINAL_VALUE])
+        return self.get_x_nominal()
 
     @property
     def x_nominal_free(self) -> List:
-        """Parameter table nominal values, for parameters to estimate."""
-        estimated = self.parameter_df[self.parameter_df[ESTIMATE] != 0]
-        return list(estimated[NOMINAL_VALUE])
+        """Parameter table nominal values, for free parameters."""
+        return self.get_x_nominal(fixed=False)
 
     @property
-    def lb(self) -> List:
-        """Parameter table lower bounds"""
-        return list(self.parameter_df[LOWER_BOUND])
-
-    @property
-    def ub(self) -> List:
-        """Parameter table upper bounds"""
-        return list(self.parameter_df[UPPER_BOUND])
+    def x_nominal_fixed(self) -> List:
+        """Parameter table nominal values, for fixed parameters."""
+        return self.get_x_nominal(free=False)
 
     @property
     def x_nominal_scaled(self) -> List:
         """Parameter table nominal values with applied parameter scaling"""
-        return list(parameters.map_scale(self.parameter_df[NOMINAL_VALUE],
-                                         self.parameter_df[PARAMETER_SCALE]))
+        return self.get_x_nominal(scaled=True)
 
     @property
     def x_nominal_free_scaled(self) -> List:
         """Parameter table nominal values with applied parameter scaling,
-        for parameters to estimate."""
-        estimated = self.parameter_df[self.parameter_df[ESTIMATE] != 0]
-        return list(parameters.map_scale(estimated[NOMINAL_VALUE],
-                                         estimated[PARAMETER_SCALE]))
+        for free parameters."""
+        return self.get_x_nominal(fixed=False, scaled=True)
+
+    @property
+    def x_nominal_fixed_scaled(self) -> List:
+        """Parameter table nominal values with applied parameter scaling,
+        for fixed parameters."""
+        return self.get_x_nominal(free=False, scaled=True)
+
+    def get_lb(self, free: bool = True, fixed: bool = True,
+               scaled: bool = False):
+        v = self.parameter_df[LOWER_BOUND]
+        if scaled:
+            v = parameters.map_scale(v, self.parameter_df[PARAMETER_SCALE])
+        return self._to_mask(v, free=free, fixed=fixed)
+
+    @property
+    def lb(self) -> List:
+        """Parameter table lower bounds."""
+        return self.get_lb()
 
     @property
     def lb_scaled(self) -> List:
         """Parameter table lower bounds with applied parameter scaling"""
-        return list(parameters.map_scale(self.parameter_df[LOWER_BOUND],
-                                         self.parameter_df[PARAMETER_SCALE]))
+        return self.get_lb(scaled=True)
+
+    def get_ub(self, free: bool = True, fixed: bool = True,
+               scaled: bool = False):
+        v = self.parameter_df[UPPER_BOUND]
+        if scaled:
+            v = parameters.map_scale(v, self.parameter_df[PARAMETER_SCALE])
+        return self._to_mask(v, free=free, fixed=fixed)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Argument 'free' passed by position and keyword in method call](https://app.codacy.com/manual/dweindl/PEtab/pullRequest?prid=4968439)